gremlin-test/src/main/java/org/apache/tinkerpop/gremlin/process/computer/bulkdumping/BulkExportVertexProgramTest.java

```diff
@@ -0,0 +1,99 @@
+
+/**
+ * @author Daniel Kuppitz (http://gremlin.guru)
+ */
+public class BulkExportVertexProgramTest extends AbstractGremlinProcessTest {
+
+    @Test
+    @LoadGraphWith(MODERN)
+    public void shouldExportCsvFile() throws Exception {
+        if (graphProvider.getGraphComputer(graph).features().supportsResultGraphPersistCombination(GraphComputer.ResultGraph.NEW, GraphComputer.Persist.EDGES)) {
+            g.V().hasLabel("person").match(
+                    __.as("person").values("name").as("name"),
+                    __.as("person").values("age").as("age"),
+                    __.as("person").outE("created").count().as("projects")
+            ).select("person").
+                    coalesce(__.out("created"), __.identity()).
+                    coalesce(__.in("created"), __.identity()).
+                    groupCount("m").by(__.select("person", "name", "age", "projects")).cap("m").unfold().as("kv").
+                    select(keys).select("person").as("person").
+                    select("kv").select(keys).select("name").as("name").
+                    select("kv").select(keys).select("age").as("age").
+                    select("kv").select(keys).select("projects").as("projects").
+                    select("kv").select(values).map(__.union(__.identity(), __.constant(-1)).sum()).as("coworkers").
+                    select("person", "name", "age", "projects", "coworkers").
+                    program(BulkDumperVertexProgram.build().create(graph)).iterate();
+
+            final Configuration config = graph.configuration();
+            final String outputLocation = config.getString("gremlin.hadoop.outputLocation");
+            final String scriptFileName =
+                    TestHelper.generateTempFileFromResource(ScriptResourceAccess.class, "script-csv-export.groovy", "").
+                            getAbsolutePath().replace('\\', '/');
+
+            config.setProperty("gremlin.hadoop.inputLocation", outputLocation + "/~g");
+            config.setProperty("gremlin.hadoop.outputLocation", outputLocation + "-csv");
+            config.setProperty("gremlin.hadoop.graphReader", "org.apache.tinkerpop.gremlin.hadoop.structure.io.gryo.GryoInputFormat");
+            config.setProperty("gremlin.hadoop.graphWriter", "org.apache.tinkerpop.gremlin.hadoop.structure.io.script.ScriptOutputFormat");
+            config.setProperty("gremlin.hadoop.scriptOutputFormat.script", scriptFileName);
+
+            graph.compute().program(BulkExportVertexProgram.build().keys("name", "age", "projects", "coworkers").create(graph)).submit().get();
+
+            final Set<String> lines = new HashSet<>();
+            lines.add("josh,32,2,3");
+            lines.add("marko,29,1,2");
+            lines.add("peter,35,1,2");
+            lines.add("vadas,27,0,0");
+
+            final File output = new File(outputLocation + "-csv/~g");
+            for (final File f : output.listFiles()) {
+                if (f.getName().startsWith("part-")) {
```

Comment:
Again, more Hadoop stuff that should not be here. Also, this is file handling, please look into `Storage`.